package.json

```diff
@@ -115,69 +115,88 @@
                 "title": "SFDX: Scan Current File for Performance Issues with ApexGuru"
             }
         ],
-        "configuration": {
-            "title": "Salesforce Code Analyzer",
-            "properties": {
-                "codeAnalyzer.pMD.customConfigFile": {
+        "configuration": [{
+			"title": "General",
+			"properties": {
+				"codeAnalyzer.enableV5": {
+					"type": "boolean",
+					"default": false,
+					"description": "Use Code Analyzer v5 (Beta) instead of Code Analyzer v4."
+				},
+                "codeAnalyzer.analyzeOnSave.enabled": {
+                    "type": "boolean",
+                    "default": false,
+                    "description": "Scan files on save automatically."
+                },
+                "codeAnalyzer.analyzeOnOpen.enabled": {
+                    "type": "boolean",
+                    "default": false,
+                    "description": "Scan files on open automatically."
+                },
+				"codeAnalyzer.apexGuru.enabled": {
+                    "type": "boolean",
+                    "default": false,
+                    "description": "Discover critical problems and performance issues in your Apex code with ApexGuru, which analyzes your Apex files for you. This feature is in a closed pilot; contact your account representative to learn more."
+                }
+			}
+		}, {
+			"title": "Code Analyzer v5 (Beta)",
+			"properties": {
+				"codeAnalyzer.ruleSelectors": {
+					"type": "string",
+					"default": "Recommended",
+					"markdownDescription": "Specifies Rule Selectors to use during default execution of Code Analyzer v5 (Beta). Use commas for unions (e.g., \"Security,Performance\") and colons for intersections (e.g., \"pmd:Security\").",
+					"description": "Specifies Rule Selectors to use during default execution of Code Analyzer v5 (Beta)."
+				}
+			}
+		}, {
+			"title": "Code Analyzer v4",
+			"properties": {
+				"codeAnalyzer.pMD.customConfigFile": {
                     "type": "string",
                     "default": "",
-                    "description": "Replace Code Analyzer's default PMD config file, choose a custom file."
+                    "description": "(v4 only) Replace Code Analyzer's default PMD config file, choose a custom file."
```

Comment:
Nit: Is "(v4 only)" needed? we already have this under the v4 section.

Reply:
Using the search can hide the section dividers, so having it in the description is a fallback.